task/sql-tasks.js

```diff
@@ -281,7 +490,25 @@ async function task_1_21(db) {
  * @return {array}
  */
 async function task_1_22(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+        SELECT DISTINCT
+            CompanyName, 
+            ProductName, 
+            OrderDetails.UnitPrice as PricePerItem
+        FROM Customers
+            INNER JOIN Orders ON Customers.CustomerID = Orders.CustomerID
+            INNER JOIN OrderDetails ON Orders.OrderID = OrderDetails.OrderID
+            INNER JOIN Products ON Products.ProductID = OrderDetails.ProductID
+        WHERE OrderDetails.UnitPrice IN (
+            SELECT MAX(UnitPrice)
+            FROM Customers cust
+                INNER JOIN Orders ON Customers.CustomerID = Orders.CustomerID
+                INNER JOIN OrderDetails ON Orders.OrderID = OrderDetails.OrderID
+            WHERE cust.CompanyName = Customers.CompanyName

```

Comment:
company name doesn't uniquely identify customer. also this column doesn't have an index so it's also slow.